networkx/algorithms/centrality/laplacian.py

```diff
@@ -0,0 +1,145 @@
+
+    G : graph
+        A networkx graph
+
+    normalized : bool (default = True)
+        If True the centrality score is scaled so the sum over all nodes is 1.
+        If False the centrality score for each node is the drop in Laplacian
+        energy when that node is removed.
+
+    nodelist : list, optional (default = None)
+        The rows and columns are ordered according to the nodes in nodelist. If nodelist is None, then the ordering is produced by G.nodes().
+
+    weight: string or None, optional (default=`weight`)
+        Optional parameter for the Laplacian matrix calculation. The edge data key used to compute each value in the matrix. If None, then each edge has weight 1.
+
+    walk_type : string or None, optional (default=None)
+        Optional parameter for the Laplacian matrix calculation. If None, P is selected depending on the properties of the graph. Otherwise is one of `random`, `lazy`, or `pagerank`.
+
+    alpha : real (default = 0.95)
+        Optional parameter for the Laplacian matrix calculation. (1 - alpha) is the teleportation probability used with pagerank.
+
+    Returns
+    -------
+    nodes : dictionary
+       Dictionary of nodes with Laplacian centrality as the value.
+
+    Examples
+    --------
+    >>> G = nx.Graph()
+    >>> edges = [(0, 1, 4), (0, 2, 2), (2, 1, 1), (1, 3, 2), (1, 4, 2), (4, 5, 1)]
+    >>> G.add_weighted_edges_from(edges)
+    >>> sorted((v, f"{c:0.2f}") for v, c in laplacian_centrality(G).items())
+    [(0, '0.70'), (1, '0.90'), (2, '0.28'), (3, '0.22'), (4, '0.26'), (5, '0.04')]
+
+    Notes
+    -----
+    The algorithm is implemented based on [1] with an extension to directed graphs
+    using the `nx.directed_laplacian_matrix` function.
+
+    Raises
+    ------
+    NetworkXPointlessConcept
+        If the graph `G` is the null graph.
+
+    References
+    ----------
+    .. [1] Qi, X., Fuller, E., Wu, Q., Wu, Y., and Zhang, C.-Q. (2012).
+    Laplacian centrality: A new centrality measure for weighted networks.
+    Information Sciences, 194:240-253.
+    https://math.wvu.edu/~cqzhang/Publication-files/my-paper/INS-2012-Laplacian-W.pdf
+    """
+    import numpy as np
+    import scipy as sp
+    import scipy.linalg  # call as sp.linalg
+    import scipy.sparse  # call as sp.sparse
+
+    def eigh_f(A):
+        return sp.linalg.eigh(A.toarray(), eigvals_only=True)
+
+    if len(G) == 0:
+        raise nx.NetworkXPointlessConcept(
+            "cannot compute centrality for the null graph"
+        )
+
+    if nodelist != None:
+        nodeset = set(G.nbunch_iter(nodelist))
+        if len(nodeset) != len(nodelist):
+            raise nx.NetworkXError(
+                "nodelist contains duplicate nodes or nodes not in G"
+            )
+        full_nodelist = nodelist + [n for n in G if n not in nodeset]
+    else:
+        nodeset = set(G.nbunch_iter(nodelist))
+        full_nodelist = nodeset
+
+    if G.is_directed():
+        lap_matrix = sp.sparse.csr_matrix(
```

Comment:
NetworkX now prefers SciPy's sparse arrays to sparse matrices, so this wrapper should be removed (sparse arrays are returned by default for NX 3.0).

Note this may also necessitate updating indexing semantics, e.g. in `new_diag` below.

Reply:
As I see only the NX  2.8.7 is available and NX 3 dev version. Please correct me if I am wrong, but as I see it is possible to merge the current version of this code into the current version of NX and then I can rewrite the code and merge the new version to the NX 3? 

I think this can be a viable option as this feature will be available in both versions of NX. What do you think @rossbar ?

Reply:
sparse arrays are preferred now -- We don't need to wait for NX3.0.